_posts/2020-07-12-ray-tune.md

```diff
@@ -0,0 +1,264 @@
+---
+layout: post
+title: Dask-ML and Ray have the same features for model selection
+tagline: Modern hyperparameter optimizations, Scikit-learn support, framework support and scaling to many machines.
+author: <a href="https://stsievert.com">Scott Sievert</a> (University of Wisconsin–Madison)
+tags: [machine-learning, dask-ml, dask, ray]
+theme: twitter
+---
+
+{% include JB/setup %}
+
+Hyperparameter optimization is the process of deducing machine learning model
+parameters that can't be learned from data. These "hyperparameters" have to be
+given at model initialization. An example is with the `alpha` parameter in
+Scikit-learn's [Ridge]. For a more complete overview, see "[How to Use t-SNE
+Effectively][tsne]" or "[Tuning the hyper-parameters of an estimator][tuning]."
+
+There's a host of libraries and frameworks out there to address this problem,
+including a popular implementation [in Scikit-learn][skl-ms]. This interface
+has been mirrored [in Dask-ML][dml-ms] and [auto-sklearn], both of which offer
+advanced hyperparameter optimization features.  Dask-ML's implementation is
+illustrated in "[Better and faster hyperparameter optimization with
+Dask][db-bf]."
+
+[auto-sklearn]:https://automl.github.io/auto-sklearn/master/
+
+[Ray] recently provided a Scikit-learn interface to [Ray Tune].  [The
+introduction][1] of this library, tune-sklearn, exaggerates it's novelty:
+
+[Ray]:https://docs.ray.io
+[Ray Tune]:https://docs.ray.io/en/master/tune.html
+
+[dml-ms]:https://ml.dask.org/hyper-parameter-search.html
+[skl-ms]:https://scikit-learn.org/stable/modules/grid_search.html
+[tsne]:https://distill.pub/2016/misread-tsne/
+[tuning]:https://scikit-learn.org/stable/modules/grid_search.html
+
+[Ridge]:https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.Ridge.html#sklearn.linear_model.Ridge
+[1]:https://medium.com/distributed-computing-with-ray/gridsearchcv-2-0-new-and-improved-ee56644cbabf
+
+> Cutting edge hyperparameter tuning techniques (bayesian optimization, early
+> stopping, distributed execution) can provide significant speedups over grid
+> search and random search.
+>
+> However, the machine learning ecosystem is missing a solution that provides
+> users with the ability to leverage these new algorithms while allowing users
+> to stay within the Scikit-Learn API. In this blog post, we introduce
+> tune-sklearn to bridge this gap. Tune-sklearn is a drop-in replacement for
+> Scikit-Learn’s model selection module with state-of-the-art optimization
+> features.
+>
+> —[GridSearchCV 2.0 — New and Improved][1]
+
+This exaggerates Ray's novelty because Dask-ML and auto-sklearn also provide
+"users with the ability to leverage these new algorithms while allowing users
+to stay within the Scikit-Learn API." Ray is not the first project to "bridge
+this gap."
+
+That is, Dask-ML offers the same features that Ray's tune-sklearn offers:
```

Comment:
I think can be removed if my suggestion is accepted.

Reply:
Or perhaps, something like "Let's step through each of the 4 bullets in the tune-sklearn introduction"

(edit: I see you have something like that below, with "Let's look at each feature..."